sdk/python/kfp/dsl/_pipeline.py

```diff
@@ -27,18 +29,29 @@
 # pipeline definition.
 _pipeline_decorator_handler = None
 
-
-def pipeline(name: Optional[str] = None,
-             description: Optional[str] = None,
-             pipeline_root: Optional[str] = None):
+class PipelineExecutionMode(enum.Enum):
+  # Compile to Argo YAML without support for metadata-enabled components.
+  V1_LEGACY = 1
+  # Compiles to Argo YAML with support for metadata-enabled components.
+  # Pipelines compiled using this mode aim to be compatible with v2 semantics.
+  V2_COMPATIBLE = 2
+  # Compiles to KFP v2 IR for execution using the v2 engine.
+  # This option is unsupported right now.
+  V2_ENGINE = 3
+
+
+def pipeline(
+    name: Optional[str] = None,
+    description: Optional[str] = None,
+    pipeline_root: Optional[str] = None):
   """Decorator of pipeline functions.
 
   Example
     ::
 
       @pipeline(
-        name='my awesome pipeline',
-        description='Is it really awesome?'
+        name='MyPipeline',
```

Comment:
nit: `MyPipeline` is not a v2 compatible name, use `my-pipeline` so that the example works on both v1 and v2 engines.

https://github.com/kubeflow/pipelines/blob/9bc63f59b27886a84bfcf4ece1062d489d06e0f5/sdk/python/kfp/v2/compiler/compiler_utils.py#L75-L80

Reply:
Done.